docs/features/software-catalog/descriptor-format.md

```diff
@@ -0,0 +1,295 @@
+
+```js
+{
+  "apiVersion": "backstage.io/v1beta1",
+  "kind": "Component",
+  "metadata": {
+    "annotations": {
+      "backstage.io/managed-by-location": "file:/tmp/component-info.yaml",
+      "example.com/service-discovery": "artistweb",
+      "circleci.com/project-slug": "gh/example-org/artist-website"
+    },
+    "description": "The place to be, for great artists",
+    "etag": "ZjU2MWRkZWUtMmMxZS00YTZiLWFmMWMtOTE1NGNiZDdlYzNk",
+    "generation": 1,
+    "labels": {
+      "system": "public-websites"
+    },
+    "name": "artist-web",
+    "uid": "2152f463-549d-4d8d-a94d-ce2b7676c6e2"
+  },
+  "spec": {
+    "lifecycle": "production",
+    "owner": "artist-relations@example.com",
+    "type": "website"
+  }
+}
+```
+
+The corresponding descriptor file that generated it may look as follows:
+
+```yaml
+apiVersion: backstage.io/v1beta1
+kind: Component
+metadata:
+  name: artist-web
+  description: The place to be, for great artists
+  labels:
+    system: public-websites
+  annotations:
+    example.com/service-discovery: artistweb
+    circleci.com/project-slug: gh/example-org/artist-website
+spec:
+  type: website
+  lifecycle: production
+  owner: artist-relations@example.com
+```
+
+The root fields `apiVersion`, `kind`, `metadata`, and `spec` are part of the
+_envelope_, defining the overall structure of all kinds of entity. Likewise,
+some metadata fields like `name`, `labels`, and `annotations` are of
+special significance and have reserved purposes and distinct shapes.
+
+See below for details about these fields.
+
+## Common to All Kinds: The Envelope
+
+The root envelope object has the following structure.
+
+### `apiVersion` and `kind` [required]
+
+The `kind` is the high level entity type being described. The
+[Backstage system model](https://github.com/spotify/backstage/issues/390) describes
+a number of core kinds that plugins can know of and understand, but an organization
+using Backstage is free to also add entities of other kinds to the catalog.
+
+The `apiVersion` is the version of specification format for that particular
+entity that the specification is made against. The version is used for being able to
+evolve the format, and the tuple of `apiVersion` and `kind` should be enough
+for a parser to know how to interpret the contents of the rest of the data.
+
+Backstage specific entities have an `apiVersion` that is prefixed with
+`backstage.io/`, to distinguish them from other types of object that share
+the same type of structure. This may be relevant when co-hosting these
+specifications with e.g. kubernetes object manifests, or when an organization
+adds their own specific kinds of entity to the catalog.
+
+Early versions of the catalog will be using beta versions, e.g. `backstage.io/v1beta1`,
+to signal that the format may still change. After that, we will be using
+`backstage.io/v1` and up.
+
+### `metadata` [required]
+
+A structure that contains metadata about the entity, i.e. things that aren't directly
```

Comment:
I think it is becoming increasingly difficult to know the difference between metadata and spec.
In this PR, why `lifecycle` is spec and not metadata? 
Maybe things should be divided in some other way or not divided at all and all be inside `spec`?

Reply:
it is indeed weird that this thing is so important to be here, same for lifecycle.
As I said, I do not really have an answer for "owner" that in general makes sense so I'll just put there the name of the company.